src/java.base/share/classes/sun/security/x509/OtherName.java

```diff
@@ -203,16 +204,12 @@ public boolean equals(Object other) {
     }
 
     /**
-     * Returns the hash code for this OtherName.
-     *
-     * @return a hash code value.
+     * {@return the hash code for this OtherName}
      */
+    @Override
     public int hashCode() {
         if (myhash == -1) {
-            myhash = 37 + oid.hashCode();
-            for (int i = 0; i < nameValue.length; i++) {
-                myhash = 37 * myhash + nameValue[i];
-            }
+            myhash = 37 + oid.hashCode() + Arrays.hashCode(nameValue);
```

Comment:
I think you can drop 37 here. Either that, or use `37 * oid.hashCode()`.

Reply:
You think it was a typo (+ instead of *) in the original code?

Reply:
The original code calculated `37*(37+oid.hashCode)+Arrays.hashCode`; since you already dropped the multiplication by 37, you can also drop the addition.

Reply:
Thanks for noticing this difference!

Yes, the initial value would've been exponentiated (^ or ** in your notation). However, I note that the original code computed this:

if nameValue.length > 0

  (37^nameValue.length) * oid.hashCode() + 37 + Arrays.hashCode(nameValue)

otherwise

  37 + oid.hashCode()

Well, it would've computed that, had the multiplier constant been 37 in Arrays.hashCode, which it is not; it's 31.

Since we cannot achieve absolute fidelity with the old code, I suggest we do this instead, what do you think?

        public int hashCode() {
        if (myhash == -1) {
            myhash = Arrays.deepHashCode(new Object[]{oid, nameValue});
        }
        return myhash;

Reply:
Correction. If the multiplier were 31, then it would've been this:

    return ( 30 + oid.hashCode() ) * 31**nameValue.length + Arrays.hashCode(nameValue);
              ^                       ^

@djelinski, do you think we should use this instead?

    return Arrays.deepHashCode(new Object[]{oid, nameValue});

Reply:
I think `oid.hashCode() + Arrays.hashCode(nameValue)` would have been good enough, but deepHashCode is also acceptable.

Reply:
> I think `oid.hashCode() + Arrays.hashCode(nameValue)` would have been good enough, but deepHashCode is also acceptable.

Fixed as you suggested, in 3139043370f.